messaging/mqtt/publisher.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mqtt
+
+import (
+	"errors"
+	"time"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/mainflux/mainflux/messaging"
+)
+
+var _ messaging.Publisher = (*publisher)(nil)
+
+var errPublish = errors.New("failed to publish")
+
+type publisher struct {
+	client  mqtt.Client
+	timeout time.Duration
+}
+
+// NewPublisher returns a new MQTT message publisher.
+func NewPublisher(address string, timeout time.Duration) (messaging.Publisher, error) {
+	client, err := newClient(address, timeout)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := publisher{
+		client:  client,
+		timeout: timeout,
+	}
+	return ret, nil
+}
+
+func (pub publisher) Publish(topic string, msg messaging.Message) error {
+	token := pub.client.Publish(topic, qos, false, msg.Payload)
+	ok := token.WaitTimeout(pub.timeout)
+	if ok && token.Error() != nil {
```

Comment:
What if `token.Error() != nil` in the previous line?

Reply:
dont you like this `if token.Wait() && token.Error() != nil {` better ?